### PR TITLE
merge eunit and region flag

### LINKED ISSUE #459: Eunit flag Broken
Someone messed up when implementing #422. The lines below are actually non-sense because they overwrite the e-unit criterion. 

https://github.com/marrink-lab/vermouth-martinize/blob/aa53b3e3ca734032518bd56b3b8dad0831653612/bin/martinize2#L737-L741

### PR DESCRIPTION
**Pull request by fgrunewald**
This fixes #459 at least partially. As stated in the issue there is still some ambiguity as to what exactly should happen if a resid region is defined for a molecule with more than one chain and same resids per chain

### CODE REVIEW

File: bin/martinize2 [hunk 1]
```diff
@@ -442,14 +442,11 @@ def entry():
...
-                          choices=['chain', 'molecule', 'all'],
                           help=('Establish what is the structural unit for the '
                                 'elastic network. Bonds are only created within'
-                                ' a unit.'))
-    rb_group.add_argument('-region', dest='rb_region', default='molecule',
-                          help=('Apply rubber bands only within the specified regions. '
-                                'The format is <start_resid_1>:<end_resid_1>, <start_resid_2>:<end_resid_2>...'
-                                'Start and end resids are included'))
+                                ' a unit. Options are molecule, chain, all, or a'
+                                'specified region defined by resids, with follwing'
```
Comment (pckroon) [inline bin/martinize2:448]:
```suggestion
                                'specified region defined by resids, with following'
```

File: vermouth/tests/data/integration_tests/tier-1/EN_chain/README [hunk 2]
```diff
@@ -0,0 +1,11 @@
+### DETAILS: 
+
+# Human Insluin HOMRO dimer
```
Comment (pckroon) [inline vermouth/tests/data/integration_tests/tier-1/EN_chain/README:3]:
```suggestion
# Human Insulin HOMRO dimer
```

File: vermouth/tests/data/integration_tests/tier-1/EN_chain/README [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# 2 chains which are connecte by disulphide bridge
```
Comment (pckroon) [inline vermouth/tests/data/integration_tests/tier-1/EN_chain/README:4]:
```suggestion
# 2 chains which are connected by disulphide bridge
```

File: vermouth/tests/data/integration_tests/tier-1/EN_region/README [hunk 4]
```diff
@@ -0,0 +1,13 @@
+### DETAILS: 
+
+# Human Insluin HOMRO dimer
```
Comment (pckroon) [inline vermouth/tests/data/integration_tests/tier-1/EN_region/README:3]:
```suggestion
# Human Insulin HOMRO dimer
```

File: vermouth/tests/data/integration_tests/tier-1/EN_region/README [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+# 2 chains which are connecte by disulphide bridge
```
Comment (pckroon) [inline vermouth/tests/data/integration_tests/tier-1/EN_region/README:4]:
```suggestion
# 2 chains which are connected by disulphide bridge
```

**Review by pckroon — CHANGES_REQUESTED** [review]
Few typos, nothing major.

I am starting to have doubt to what's the desired behaviour though. Is it not desirable to be able to combine `-eunit chain` with `-region 200:400`? The behaviour would be that EB only get made where both conditions are true.
Also, CLI nitpick, when I provide an `-eunit` I'd argue the `-elastic` is implied, but that can be shelved for a while.

**Comment by fgrunewald** [comment]
As far as the behaviour goes the `eunit=chain` is really just an edgy edge case. It only comes to be important when two chains, which are labelled as such are connected by a disulphide bridge. Any thing else will end as two separate molecules or one molecule. Thus I think it is fine having the eunit flag being either all, molecule, chain, or residrange.

**Comment by pckroon** [comment]
Then maybe chain + region was a poorly chosen example. 
How about molecule + region? You may want an elastic network only within a molecule/chain, but only between resids 200-400. So if you have 2 homodimers that are close together you only want EN between resid 200-400, but not between the two dimers. I think that's not possible with the current CLI suggestion

**Comment by fgrunewald** [comment]
@Tsjerk any thoughts on this

**Review by pckroon — APPROVED** [review]
Ok, that works for me. I'll open a fresh issue.

**Pull request merged**